modules/phase_field/src/kernels/KKSMultiPhaseConcentration.C

```diff
@@ -0,0 +1,106 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "KKSMultiPhaseConcentration.h"
+
+template<>
+InputParameters validParams<KKSMultiPhaseConcentration>()
+{
+  InputParameters params = validParams<Kernel>();
+  params.addClassDescription("KKS multi-phase model kernel to enforce (c = h1*c1 + h2*c2 + h3*c3 +.. The non-linear variable of this kernel is cn, the final phase concenration in the list.");
+  params.addRequiredCoupledVar("cj", "Array of phase concentrations cj. Place in same order as hj_names!");
+  params.addRequiredCoupledVar("c", "Physical concentration");
+  params.addCoupledVar("etas", "Order parameters for all phases");
+  params.addRequiredParam<std::vector<MaterialPropertyName> >("hj_names", "Switching Function Materials that provide h(eta_1, eta_2,...)");
+  return params;
+}
+
+// Phase interpolation func
+KKSMultiPhaseConcentration::KKSMultiPhaseConcentration(const InputParameters & parameters) :
+    DerivativeMaterialInterface<Kernel>(parameters),
+    _ncj(coupledComponents("cj")),
+    _cjs(_ncj),
+    _cjs_var(_ncj),
+    _c(coupledValue("c")),
+    _c_var(coupled("c")),
+    _hj_names(getParam<std::vector<MaterialPropertyName> >("hj_names")),
+    _nhj(_hj_names.size()),
+    _prop_hj(_nhj),
+    _num_etas(coupledComponents("etas")),
+    _eta_names(_num_etas),
+    _eta_vars(_num_etas),
+    _prop_dhjdetai(_nhj)
+{
+  //Check to make sure the the number of cj's is the same as the number of hj's
+  if (_ncj != _nhj)
+    mooseError("Need to pass in as many hj_names as cj_names in KKSMultiPhaseConcentration " << name());
+  //Check to make sure the the number of etas is the same as the number of hj's
+  if (_num_etas != _nhj)
```

Comment:
I'd just set _one_ member variable `_nj` and check against `.size()` of the other vectors. There is no need to store the same value multiple times.